packages/modal/src/presenters/ModalPresenter.js

```diff
@@ -0,0 +1,109 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import cx from "classnames";
+
+import ModalHeaderPresenter from "./ModalHeaderPresenter";
+import "./ModalPresenter.scss";
+
+export default class ModalPresenter extends Component {
+  static propTypes = {
+    /**
+     * Supports adding any dom content to the body of the modal
+     */
+    children: PropTypes.node,
+    /**
+     * ARIA label attribute for the close button if/when headerChildren
+     * are not utilized
+     */
+    closeButtonAriaLabel: PropTypes.string,
+    /**
+     * Supports adding any dom content to the header of the modal
+     */
+    headerChildren: PropTypes.node,
+    /**
+     * Triggers when one clicks the close button
+     */
+    onCloseClick: PropTypes.func,
+    /**
+     * Triggers when one clicks the overlay behind the modal
+     */
+    onOverlayClick: PropTypes.func,
+    /**
+     * Triggers when one clicks the modal window
+     */
+    onWindowClick: PropTypes.func,
+    /**
+     * Modal is visible when true
+     */
+    open: PropTypes.bool,
+    /**
+     * Title of the modal
+     */
+    title: PropTypes.string,
+    /**
+     * Style of the modal shell
+     */
+    type: PropTypes.string
+  };
+
+  setScrolling = element => {
+    this.hasScrolling = element.scrollHeight > element.clientHeight;
+  };
+
+  render() {
+    const {
+      children,
+      closeButtonAriaLabel,
+      headerChildren,
+      onCloseClick,
+      onOverlayClick,
+      onWindowClick,
+      open,
+      title,
+      type
+    } = this.props;
+
+    const typeKey = type || "standard";
+
+    const windowClasses = cx([
+      "hig__modal-V1__window",
+      `hig__modal-V1__window--${typeKey}`
```

Comment:
> I'm not a fan of using string interpolation in cases where a dictionary can be used.
>
> The first issue is that it couples the class name to a prop value. This is inflexible and will have to be completely rewritten when we migrate to CSS-in-JS. We should be able to change the prop values or the class names independently of each other.
>
> This also allows consumers to use the API in unintended ways. If the `style` prop is set to `standard my-custom-class`, then `hig__modal-V1__window--standard my-custom-class` is added.
> 
> Example: https://github.com/Autodesk/hig/blob/development/packages/tabs/src/presenters/TabsPresenter.js#L9-L13

https://github.com/Autodesk/hig/pull/1092#discussion_r208756578

Reply:
Thanks for the example, updated in 9be9bfd3